src/Fixer/PhpUnit/PhpUnitStrictFixer.php

```diff
@@ -99,26 +99,40 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
         foreach ($this->configuration['assertions'] as $methodBefore) {
             $methodAfter = self::$assertionMap[$methodBefore];
+            $sequencesToFind = array(
+                array(
+                    array(T_VARIABLE, '$this'),
+                    array(T_OBJECT_OPERATOR, '->'),
+                    array(T_STRING, $methodBefore),
+                    '(',
+                ),
+                array(
+                    array(T_STATIC, 'static'),
+                    array(T_DOUBLE_COLON, '::'),
+                    array(T_STRING, $methodBefore),
+                    '(',
+                ),
+                array(
+                    array(T_STRING, 'self'),
+                    array(T_DOUBLE_COLON, '::'),
+                    array(T_STRING, $methodBefore),
+                    '(',
+                ),
+            );
 
-            for ($index = 0, $limit = $tokens->count(); $index < $limit; ++$index) {
-                $sequence = $tokens->findSequence(
-                    array(
-                        array(T_VARIABLE, '$this'),
-                        array(T_OBJECT_OPERATOR, '->'),
-                        array(T_STRING, $methodBefore),
-                        '(',
-                    ),
-                    $index
-                );
+            foreach ($sequencesToFind as $sequenceToFind) {
+                for ($index = 0, $limit = $tokens->count(); $index < $limit; ++$index) {
```

Comment:
I believe that there is no need to iterate whole tokens collection 3x, originally we did only once.

Reply:
Algorithm optimized